test/CMakeLists.txt

```diff
@@ -147,6 +180,25 @@ function(add_test_u TEST_NAME)
                          LABELS "unit")
 endfunction(add_test_u)
 
+# Add test that also checks post-processing scripts: sampling
+function(add_test_pps TEST_NAME FORMAT NSTEPS)
+    setup_pp_test()
+    set(ADDL_RUNTIME_OPTIONS "sampling.output_format=${FORMAT} time.plot_interval=-1 time.checkpoint_interval=-1 time.max_step=${NSTEPS}")
+    # need to somehow only install the requirements once, prior to testing, and then reference their location in testing
+    add_test("${TEST_NAME}_sampling_${FORMAT}" sh -c "${MPI_COMMANDS} ${CMAKE_BINARY_DIR}/${amr_wind_exe_name} ${MPIEXEC_POSTFLAGS} \
+            \ ${CURRENT_TEST_BINARY_DIR}/${TEST_NAME}.inp ${RUNTIME_OPTIONS} ${ADDL_RUNTIME_OPTIONS} > ${TEST_NAME}_sampling_${FORMAT}.log && \
+            \ ${Python_EXECUTABLE} -m venv .venv && source .venv/bin/activate && .venv/bin/pip3 install ${PYTHON_REQUIREMENTS} && \
```

Comment:
We shouldn't be trying to fulfill dependencies in CMake.

Reply:
good call